src/rasterstats/main.py

```diff
@@ -254,7 +260,10 @@ def gen_zonal_stats(
 
             if add_stats is not None:
                 for stat_name, stat_func in add_stats.items():
-                    feature_stats[stat_name] = stat_func(masked)
+                    try:
+                        feature_stats[stat_name] = stat_func(masked, prop_dic)
+                    except:
+                        feature_stats[stat_name] = stat_func(masked)
```

Comment:
I'd like to avoid a catch-all exception here.

What about 

```python
                    try:
                        feature_stats[stat_name] = stat_func(masked, feat['properties'])
                    except TypeError:
                        # backwards compatible with single-argument function
                        feature_stats[stat_name] = stat_func(masked) 
```